Lombiq.HelpfulLibraries.LinqToDb/LinqToDbQueryExecutor.cs

```diff
@@ -0,0 +1,59 @@
+using Dapper;
+using LinqToDB;
+using LinqToDB.Data;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Threading.Tasks;
+using YesSql;
+
+namespace Lombiq.HelpfulLibraries.LinqToDb
+{
+    public static class LinqToDbQueryExecutor
+    {
+        public static async Task<IEnumerable<TResult>> LinqQueryAsync<TResult>(
+            this ISession session,
+            Func<ITableAccessor, IQueryable> query)
+        {
+            var transaction = await session.DemandAsync();
+            var convertedSql = ConvertSqlToDialect(session, transaction, query);
+
+            return await transaction.Connection.QueryAsync<TResult>(convertedSql, transaction: transaction);
+        }
+
+        private static string ConvertSqlToDialect(
+            ISession session,
+            IDbTransaction transaction,
+            Func<ITableAccessor, IQueryable> query)
+        {
+            // Generate aliases for final projection.
+            LinqToDB.Common.Configuration.Sql.GenerateFinalAliases = true;
+
+            // Instantiating a linq2db connection object as it is required to start building the query. Note that it
+            // won't create an actual connection with the database yet.
+            var dataProvider = DataConnection.GetDataProvider(
+                GetDatabaseProviderName(session.Store.Dialect.Name),
+                transaction.Connection.ConnectionString);
+
+            using var linqToDbConnection = new LinqToDbConnection(
+                dataProvider,
+                transaction,
+                session.Store.Configuration.TablePrefix);
+            return query(linqToDbConnection).ToString();
+        }
+
+        private static string GetDatabaseProviderName(string dbName) =>
+            dbName switch
+            {
+                // Using explicit string instead of LinqToDB.ProviderName.SqlServer because the "System.Data.SqlClient"
+                // provider will be used causing "Could not load type System.Data.SqlClient.SqlCommandBuilder"
+                // exception. See: https://github.com/linq2db/linq2db/issues/2191#issuecomment-618450439

```

Comment:
Itt valami nyelvtani hiba van.

Reply:
Valami? :D Le kellene töltenem valami grammar plugint VS-hez?

Reply:
Lehet csak egy if hiányzik. "Using explicit string instead of LinqToDB.ProviderName.SqlServer because **if** the..."

Reply:
De neked ezt mi mutatja? Vagy csak elolvasva nem oké? Mert csekkoltam valami online grammar check tool-al és nincs baja :D
Frissítettem.

Reply:
Én ezt a fejemben lévő masszívan párhuzamos architektúrával szúrtam ki :).